src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/tasklet/AbstractCatalogBackupRestoreTasklet.java

```diff
@@ -0,0 +1,465 @@
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.tasklet;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.io.FileUtils;
+import org.geoserver.backuprestore.AbstractExecutionAdapter;
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.utils.BackupUtils;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.config.ServiceInfo;
+import org.geoserver.config.util.XStreamPersister;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.geoserver.config.util.XStreamServiceLoader;
+import org.geoserver.gwc.config.GWCConfig;
+import org.geoserver.gwc.config.GWCConfigPersister;
+import org.geoserver.gwc.config.GeoserverXMLResourceProvider;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Files;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.platform.resource.Resources.AnyFilter;
+import org.geoserver.util.Filter;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.util.Assert;
+
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * TODO
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public abstract class AbstractCatalogBackupRestoreTasklet implements Tasklet, InitializingBean {
+
+    /*
+     * 
+     */
+    protected static Map<String, Filter<Resource>> resources = new HashMap<String, Filter<Resource>>();
+
+    /*
+     * 
+     */
+    static {
+        resources.put("demo", AnyFilter.INSTANCE);
+        resources.put("images", AnyFilter.INSTANCE);
+        resources.put("logs", new Filter<Resource>() {
+
+            @Override
+            public boolean accept(Resource res) {
+                if (res.name().endsWith(".properties")) {
+                    return true;
+                }
+                return false;
+            }
+            
+        });
+        resources.put("palettes", AnyFilter.INSTANCE);
+        resources.put("plugIns", AnyFilter.INSTANCE);
+        resources.put("styles", new Filter<Resource>() {
```

Comment:
Why are styles treated as additional resources, I see there is already a dedicated step in the backup and restore flow to handle them.